docs/source/recipes/plot_11_recipe.py

```diff
@@ -0,0 +1,90 @@
+"""
+Plotting the Warming Stripes
+============================
+
+In this recipe, we will plot the `Warming Stripes (Climate Stripes)
+<https://en.wikipedia.org/wiki/Warming_stripes>`_ created by
+Professor Ed Hawkins at NCAS, University of Reading. Here we will use the
+ensemble mean of the 
+`HadCRUT.5.0.1.0 analysis gridded data
+<https://www.metoffice.gov.uk/hadobs/hadcrut5/data/current/download.html>`_ for
+the same.
+
+"""
+
+# %%
+# 1. Import cf-python and matplotlib.pyplot:
+
+import matplotlib.pyplot as plt
+
+import cf
+
+# %%
+# 2. Read the field constructs:
+temperature_data = cf.read(
+    "~/recipes/HadCRUT.5.0.1.0.analysis.anomalies.ensemble_mean.nc"
+)[0]
+print(temperature_data)
+
+# %%
+# 3. Calculate the annual mean temperature anomalies. The ``'weights=True'``
+# argument is used to take the varying lengths of months into account which
+# ensures that the calculated mean is more accurate:
+annual_temperature = temperature_data.collapse(
+    "T: mean", weights=True, group=cf.Y()
+)
+
+# %%
+# 4. Select the data from 1850 to 2022:
+period = annual_temperature.subspace(T=cf.year(cf.wi(1850, 2022)))
+
+# %%
+# 5. Calculate the global average temperature for each year:
+global_temperature = period.collapse("X: Y: mean")
+
+# %%
+# 6. Get the global average temperature and squeeze it to remove the size 1 axis:
+global_avg_temp = global_temperature.array.squeeze()
+
+# %%
+# 7. Create a normalization function that maps the interval from the minimum to 
+# the maximum temperature to the interval [0, 1] for colouring:
+norm_global = plt.Normalize(global_avg_temp.min(), global_avg_temp.max())
+
+# %%
+# 8. Set the colormap instance:
+cmap = plt.get_cmap("RdBu_r")
+
+# %%
+# 9. Create the figure and the axes for the global plot. Loop over the selected 
+# years, plot a colored vertical stripe for each and remove the axes:
+fig_global, ax_global = plt.subplots(figsize=(10, 2))
+
+for i in range(global_avg_temp.shape[0]):
+    ax_global.axvline(
+        x=i, color=cmap(norm_global(global_avg_temp[i])), linewidth=4
+    )
```

Comment:
> Do you think we should use [matplotlib.axes.Axes.axvline](https://matplotlib.org/stable/api/_as_gen/matplotlib.axes.Axes.axvline.html) with less parameters or [matplotlib.axes.Axes.fill_between](https://matplotlib.org/stable/api/_as_gen/matplotlib.axes.Axes.fill_between.html#matplotlib.axes.Axes.fill_between)?

I had a little play around with the options, and actually I think neither! From looking in the docs to see what these do, I noticed an alternative,[`axvspan`](https://matplotlib.org/stable/api/_as_gen/matplotlib.axes.Axes.axvspan.html), which is perfect for what we need, because it can be set to start at the end of the last stripe, horizontally considered, and end at the start of the next one, and so on:

```suggestion
for i in range(global_avg_temp.shape[0]):
    ax_global.axvspan(
        xmin=i-0.5,
        xmax=i+0.5,
        color=cmap(norm_global(global_avg_temp[i]))
    )
```

This means that regardless of the zoom level applied, there won't be any gaps between the stripes, e.g:

![cs_zoom_Figure_1](https://github.com/NCAS-CMS/cf-python/assets/30274190/f88f1b0c-2ef3-4668-9076-f02ff9452fa2)

Whereas `linewidth` varies according to zoom level so gaps will always be liable to appear. So this way, i.e. that in the suggestion, seems most robust to me. And the code stays simple, especially when the arguments are listed out as `xmin=`, `xmax=` to make it clear what we're plotting.

Does it seem OK to you as a solution? See also the regional stripes where I am making the equivalent suggestion.

Reply:
Ah nice! [`axvspan`](https://matplotlib.org/stable/api/_as_gen/matplotlib.axes.Axes.axvspan.html), is the perfect one that we need! Thanks Sadie!